builder/docker/step_temp_dir.go

```diff
@@ -18,7 +19,23 @@ func (s *StepTempDir) Run(state multistep.StateBag) multistep.StepAction {
 	ui := state.Get("ui").(packer.Ui)
 
 	ui.Say("Creating a temporary directory for sharing data...")
-	td, err := ioutil.TempDir("", "packer-docker")
+
+	var err error
+	var td string
+	tempenv := "TMPDIR"
+	if runtime.GOOS == "windows" {
+		tempenv = "TMP"
+	}
+	if tmpdir := os.Getenv(tempenv); tmpdir != "" {
+		td, err = ioutil.TempDir(tmpdir, "packer-docker")
+	} else {
+		var configTmpDir string
+		configTmpDir, err = packer.ConfigTmpDir()
+		if err == nil {
+			td, err = ioutil.TempDir(configTmpDir, "packer-docker")
+		}
+	}
```

Comment:
I might have missed this, but is there a default case if the environment variable and the config file are blank?